clicommand/pipeline_upload.go

```diff
@@ -107,6 +117,17 @@ var PipelineUploadCommand = cli.Command{
 			Usage:  "When true, fail the pipeline upload early if the pipeline contains secrets",
 			EnvVar: "BUILDKITE_AGENT_PIPELINE_UPLOAD_REJECT_SECRETS",
 		},
+		cli.StringFlag{
+			Name:   "signing-key",
+			Usage:  "Path to a file containing a signing key. Passing this flag enables pipeline signing. The format of the file depends on the chosen signing algorithm",
+			EnvVar: "BUILDKITE_PIPELINE_SIGNING_KEY_FILE",
+		},
+		cli.StringFlag{
+			Name:   "signing-algorithm",
+			Usage:  "Signing algorithm to use when signing parts of the pipeline. Available algorithms are: hmac-sha256",
+			Value:  "hmac-sha256",
```

Comment:
given that we're planning on adding asymmetric keys to the signing feature, it might be good to not default to anything from the start?

Reply:
Hmm, yeah, good point